pkgs/development/python-modules/sip/4.x.nix

```diff
@@ -30,6 +31,12 @@ buildPythonPackage rec {
     })
   ];
 
+  postPatch = lib.optionalString (pythonAtLeast "3.12") ''
+    substituteInPlace configure.py --replace-fail "from distutils" "from setuptools._distutils"
+  '';
+
+  propagatedBuildInputs = lib.optional (pythonAtLeast "3.12") setuptools;
```

Comment:
Is this because it uses distutils at runtime?

Reply:
I'm already forgetting the details, but yes I believe sip has a distutils based project builder that requires it at runtime

Reply:
I think it is possible that your new changes in #342948 removing the long replace will mean we must also add distutils here from some sip usages, or it may not import correctly.